p2p/src/authenticated/actors/tracker/ingress.rs

```diff
@@ -150,11 +149,6 @@ impl<E: Spawner + Metrics, C: Verifier> Reservation<E, C> {
 impl<E: Spawner + Metrics, C: Verifier> Drop for Reservation<E, C> {
     fn drop(&mut self) {
         let (peer, mut mailbox) = self.closer.take().unwrap();
-        self.context
-            .clone()
-            .with_label("reservation")
-            .spawn(move |_| async move {
-                mailbox.release(peer).await;
-            });
+        block_on(mailbox.release(peer));
```

Comment:
If the mailbox is full, I think this could cause a deadlock if there is only 1 core allocated for async execution...

Reply:
Oooooo...we could use try_send here to avoid spawning in most cases.